tests/test_flow/test_persistence_fuzz.py

```diff
@@ -221,14 +248,23 @@ def run(self, n_iterations):
                     returned_value = self.model.build_flow().get(affected_name)
                     expected_value = self.model.expected_entity_value(
                         affected_name)
-                    if make_func_change:
+
+                    # value will be different if we made a functional change and
+                    # updated entity does not change per run. If updated entity does
+                    # change, the value hash will change and everything will recompute.
+                    if make_func_change and not is_updated_entity_nondeterministic:
                         assert returned_value != expected_value
                     else:
                         assert returned_value == expected_value
 
-                    # We should have used the cached value, so no new
-                    # computation should have happened.
-                    assert len(self.model.called_entity_names()) == 0
+                    # If updated entity does not change per run, we call only nondeterministic
+                    # ancestors. It's safe to check this now since nodeterministic ancestors
+                    # will be recomputed instead of using the disk value in the later check.
+                    if not is_updated_entity_nondeterministic:
```

Comment:
Is there an assertion we can make when `is_updated_entity_nondeterministic` is true?

Reply:
I changed this to `is_updated_entity_deterministic or make_nonfunc_change` and explained why we are excluding the checks when it is false. 

Those cases are tested at the end of the method (outside the `not update_version` if conditional and inside `make_func_change` if conditional).